### DISCUSSION TITLE
Is it possible to ensure number of `expectTypeOf` which succeed are exactly `n`, like in `expect.assertions`?

### DISCUSSION THREAD

**Discussion by aarontravass** [vitest-dev/vitest#3929] [Q&A]
`expect.assertions` can be used to ensure a specific number of assertions are called and succeed. Is it possible to do the same with `expectTypeOf`?

 I would like to do something like `expectTypeOf.assertions(2)` since type assertions become difficult in callback functions.

**Comment by sheremet-va** [comment]
expectTypeOf are not called, they are statically analyzed. If it's in your code then it is always analyzed

**Reply by aarontravass** [reply]
Understood. But why does the test succeed even though the type is wrong? (it does throw the `MISMATCH` error)
![image](https://github.com/vitest-dev/vitest/assets/43901677/53f299e7-1655-4aae-b99b-5e242c67adf2)

![image](https://github.com/vitest-dev/vitest/assets/43901677/e18a50a5-19e3-48b8-8050-4caecd982ad9)

**Reply by sheremet-va** [reply]
Looks like a Vitest bug.

**Reply by aarontravass** [reply]
should I raise an issue with this bug?
